domino/domino.py

```diff
@@ -377,6 +377,8 @@ def job_start_blocking(self, poll_freq: int = 5, max_poll_time: int = 6000, **kw
         def get_job_status(job_identifier):
             status = self.job_status(job_identifier)
             self.log.info(f"Polling Job: {job_identifier} status is completed: {status['statuses']['isCompleted']}")
+            if status['statuses']['executionStatus'] == "Failed":
```

Comment:
@abhijeet2096 throwing an exception here if the status is failed so it can be handled downstream by other functions, like the airflow operator. I noticed failures were happening silenty since we just check for job completion and
`job_complete != job_success`